--- a/compiler/rustc_middle/src/ty/generics.rs
+++ b/compiler/rustc_middle/src/ty/generics.rs
@@ -1,7 +1,7 @@
 use std::ops::ControlFlow;
 
 use rustc_ast as ast;
-use rustc_data_structures::fx::FxHashMap;
+use rustc_data_structures::fx::FxIndexMap;
 use rustc_hir::def_id::DefId;
 use rustc_macros::{StableHash, TyDecodable, TyEncodable};
 use rustc_span::{Span, Symbol, bug, kw};
@@ -124,7 +124,7 @@
 
     /// Reverse map to the `index` field of each `GenericParamDef`.
     #[stable_hash(ignore)]
-    pub param_def_id_to_index: FxHashMap<DefId, u32>,
+    pub param_def_id_to_index: FxIndexMap<DefId, u32>,
 
     pub has_self: bool,
     pub has_late_bound_regions: Option<Span>,
@@ -132,8 +132,6 @@
 
 impl std::fmt::Debug for Generics {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
-        // ironically, we get this warning because of what we're trying to fix.
-        #[expect(rustc::potential_query_instability)]
         let mut stabilized_hashmap = self.param_def_id_to_index.iter().collect::<Vec<_>>();
         stabilized_hashmap.sort_by_key(|(_, v)| **v);
         f.debug_struct("Generics")
