--- a/compiler/rustc_middle/src/ty/context.rs
+++ b/compiler/rustc_middle/src/ty/context.rs
@@ -1371,8 +1371,9 @@
     }
 
     pub fn iter_local_def_id(self) -> impl Iterator<Item = LocalDefId> {
-        // Depend on the `analysis` query to ensure compilation if finished.
-        self.ensure_ok().analysis(());
+        // Create a dependency to the red node to be sure we re-execute this when the amount of
+        // definitions change.
+        self.dep_graph.read_index(DepNodeIndex::FOREVER_RED_NODE);
 
         let definitions = &self.untracked.definitions;
         gen {
